protocols/gossipsub/src/behaviour.rs

```diff
@@ -1598,28 +1598,19 @@ where
                         continue;
                     }
 
-                    // NOTE: We cannot dial any peers from PX currently as we typically will not
-                    // know their multiaddr. Until SignedRecords are spec'd this
-                    // remains a stub. By default `config.prune_peers()` is set to zero and
-                    // this is skipped. If the user modifies this, this will only be able to
-                    // dial already known peers (from an external discovery mechanism for
-                    // example).
-                    if self.config.prune_peers() > 0 {
-                        self.px_connect(px);
+                    // mesh count already discounted prune's sender
+                    let max_px_to_connect = self.config.mesh_n_low()
+                        - self.mesh.get(&topic_hash).map(|x| x.len()).unwrap_or(0);
```

Comment:
this is likely to panic on release

Also, I think you want `mesh_n_high` instead of `mesh_n_low` ? Both are fine depending on what we want, @AgeManning suggested the high one which would be interpreted as capacity more than need